rfcs/0003-dns-policy.md

```diff
@@ -0,0 +1,290 @@
+# RFC Template
+
+- Feature Name: DNSPolicy
+- Start Date: 2023-07-01
+- RFC PR: [Kuadrant/architecture#20](https://github.com/Kuadrant/architecture/pull/20)
+- Issue tracking: [Kuadrant/architecture#0000](https://github.com/Kuadrant/multicluster-gateway-controller/issues/219)
+- Labels: DNS, Load Balancing, Multi-Cluster
+    
+
+# Summary
+[summary]: #summary
+
+Provide a policy for configuring how DNS should be handed for a given gateway. Provide a mechanism for enabling DNS based load balancing.
+
+# Motivation
+[motivation]: #motivation
+
+Gateway admins, need a way to define the DNS policy for a multi-cluster gateway in order to control how much and which traffic reaches these gateways. 
+Ideally we would allow them to express a strategy that they want to use without needing to get into the details of each provider and needing to create and maintain dns record structure and individual records for all the different gateways that may be within their infrastructure.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+Allow definition of a DNSPolicy that configures load balancing to decide how traffic should be distributed across multiple gateway instances from the central control plane.
+
+## Terms
+
+- **managed listener**: This is a listener with a host backed by a DNS zone managed by the multi-cluster gateway controller
+- **hub cluster**: control plane cluster that managed 1 or more spokes
+- **spoke cluster**: a cluster managed by the hub control plane cluster. This is where gateway are instantiated
+
+
+Provide a control plane DNSPolicy API that uses the idea of direct [policy attachment](https://gateway-api.sigs.k8s.io/references/policy-attachment/#direct-policy-attachment) from gateway API that allows a load balancing strategy to be applied to the DNS records structure for any managed listeners being served by the data plane instances of this gateway. 
+The DNSPolicy also covers health checks that inform the DNS response but that is not covered in this document.
+
+Below is a draft API for what we anticipate the DNSPolicy to look like
+
+```yaml 
+
+apiVersion: kuadrant.io/v1alpha1
+kind: DNSPolicy
+spec:
+  targetRef: # defaults to gateway gvk and current namespace
```

Comment:
GVK is required, I'm afraid. Only namespace can be omitted (and assumed.) https://github.com/kubernetes-sigs/gateway-api/blob/7042cd866bc69b160ab6d04b51676e14b487505d/apis/v1alpha2/policy_types.go#L24-L41